sklearn/tree/tests/test_tree.py

```diff
@@ -1487,9 +1480,11 @@ def check_explicit_sparse_zeros(tree, max_depth=3, n_features=10):
 
     indices = np.concatenate(indices)
```

Comment:
Apparently, there a change of behaviour between sparse matrices and arrays: matrices will convert to 32 bits precision the indices if possible while the arrays will preserve the `dtype`.

Since we are not supporting 64 bits indices in the trees and this is actually not a regression, we can make sure to create 32 bits indices and the test should pass.

```suggestion
    indices = np.concatenate(indices).astype(np.int32)
    indptr = np.array(indptr, dtype=np.int32)
```

Reply:
Sorry, didn't see that CI is complaining. Will do ASAP.